kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/bytearray/ByteArrayRecordWriterProvider.java

```diff
@@ -56,15 +63,23 @@ public String getExtension() {
   @Override
   public RecordWriter getRecordWriter(final S3SinkConnectorConfig conf, final String filename) {
     return new RecordWriter() {
-      final S3OutputStream s3out = storage.create(filename, true);
+      final String adjustedFilename = getAdjustedFilename(filename, getExtension());
+      final S3OutputStream s3out = storage.create(adjustedFilename, true, ByteArrayFormat.class);
       final OutputStream s3outWrapper = s3out.wrapForCompression();
 
       @Override
       public void write(SinkRecord record) {
-        log.trace("Sink record: {}", record);
+        log.trace("Sink record with view {}: {}", recordView, record);
         try {
-          byte[] bytes = converter.fromConnectData(
-              record.topic(), record.valueSchema(), record.value());
+          byte[] bytes;
+          if (recordView instanceof HeaderRecordView) {
+            jsonConverter.configure(Collections.singletonMap("schemas.enable", false), false);
```

Comment:
This special handling for trying to encode headers as `byte[]` is awkward. I don't know if people would actually try to do that. Could we just say you can't use `ByteArrayFormat` for encoding headers? Remove it from the config recommender, etc?

Reply:
If we use `ByteArrayFormat` for the `headers.format.class` this is still the record provider that gets called. Did you mean to say `JsonFormat`? @kkonstantine may have some input on how we'd like to handle this.

Reply:
Well, this code is for handling if `ByteArrayFormat` is specified in `headers.format.class`. I don't know if that is actually useful. If not, we could just not support it and remove this one-off special case handling code.

Reply:
I see what you mean now. Yeah that may be an option. @kkonstantine

Reply:
I agree, let's remove support for headers as a byte array for now, since the use of json converter and encoding is a bit arbitrary.